### PR TITLE
Use EIP-155 signing for Ethereum contracts

### PR DESCRIPTION
**Pull request by Shadowfiend**
go-ethereum 1.10.0 has switched to denying non-EIP155 signatures by default.
Although our code uses go-ethereum's own ABI binding generation, the version
currently in our dependencies never added support for EIP155 signers, and
instead always uses Homestead signers for the transactors it provides helpers
for.

This commit bumps the keep-common version to a hotfix that enables EIP-155
signing for contract bindings. It updates generated code from keep-common to
reflect these new contract bindings, and updates Ethereum client instantiation
to resolve the chain id and pass it into the contract binding constructor
functions.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< Updated upstream
```
Comment (pdyraga) [inline go.sum:1]: Merge conflict?

Comment (pdyraga) [reply go.sum]: Dropped in https://github.com/keep-network/keep-ecdsa/pull/725/commits/1baa1eb72194531326ab2660aab344061a472893

**Comment by pdyraga** [comment]
Tested:
- client without a fix can not submit a TX to geth v1.10.1 with default start options
- client with a fix can submit a TX to geth v1.10.1 with default start options
- client with a fix can submit a TX to geth v1.10.1 with `--rpc.allow-unprotected-txs`
- client with a fix can submit a TX to (old) geth v1.9.18

**Review by pdyraga — APPROVED** [review]
LGTM but given that I was committing here, we need another approval.

**Review by nkuba — APPROVED** [review]

**Review by dimpar — APPROVED** [review]

**Pull request merged**